agent/structs/config_entry_routes.go

```diff
@@ -73,6 +81,129 @@ func (e *HTTPRouteConfigEntry) GetRaftIndex() *RaftIndex {
 	return &e.RaftIndex
 }
 
+// HTTPMatch specifies the criteria that should be
+// used in determining whether or not a request should
+// be routed to a given set of services.
+type HTTPMatch struct {
+	Headers []HTTPHeaderMatch
+	Method  HTTPMatchMethod
+	Path    HTTPPathMatch
+	Query   []HTTPQueryMatch
+}
+
+// HTTPMatchMethod specifies which type of HTTP verb should
+// be used for matching a given request.
+type HTTPMatchMethod string
+
+const (
+	HTTPMatchMethodAll     HTTPMatchMethod = ""
```

Comment:
Is this intention here that if the value is unset, then the match works for all methods? I'm picturing the scenario where a `string` is initialized with the value of `""` and matches all methods. If this is the desired implementation, the `""` makes sense, otherwise it might make more sense to have this set at `"*"`.

Reply:
My preference is to keep this as just the default. Basically, not specifying results in all HTTP methods being matched, so rather than trying to disambiguate between not specified and zero-value behavior via something like a pointer, I'd rather just use the zero-value to specify the default behavior.